catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/FilterWriter.java

```diff
@@ -49,17 +55,56 @@
+   * for linking a separate namespace using the {@code <import>} directive. To support that without
+   * modifying the XSD files, a custom {@link org.w3c.dom.ls.LSResourceResolver} is being used.
+   *
+   * <p>The {@code systemId} is the URI of the resource about to be loaded. The {@code baseURI} is
+   * the URI of the resource currently being processed, which references the {@code systemId}. By
+   * returning {@code null} in the resource resolver the {@code systemId} resource can be loaded in
+   * the default way without any custom handling. This works for {@code <include>} cases because the
+   * filter namespace documents live on the same context path, so the {@code bundleresource:<path>}
+   * URI that was constructed by OSGi via traditional resource loading can be reused. Instances of
+   * {@code systemId} will just be a simple file name in these cases, i.e. {@code sort.xsd}, without
+   * any qualifying information.
+   *
+   * <p>For {@code <import>} cases, and bringing in documents under a separate namespace, it cannot
+   * be assumed their context paths are similar. In that regard, {@code systemId}s will be fully
+   * qualified HTTP URLs and returning {@code null} will invoke default behavior that performs a
+   * network request to the open internet. Instead, the resource resolver will parse the document's
+   * name and redirect the network request by overwriting the {@code systemId} URI to point to the
+   * local version of the XSD.
+   *
    * @param validationEnabled true if all XML writing done by this {@link FilterWriter} should also
    *     be validated against the <a href="http://schemas.opengis.net/filter/2.0/filter.xsd">filter
    *     schema</a>. False otherwise.
    * @throws JAXBException if a problem occurs setting up and configuring JAXB.
+   * @see #FILTER_XSD_RESOURCE_PATH for how the root schema document is being loaded.
    */
   public FilterWriter(boolean validationEnabled) throws JAXBException {
     this.marshaller = JAXBContext.newInstance(FilterType.class).createMarshaller();
     if (validationEnabled) {
       LOGGER.info("Loading filter schemas");
       URL schemaLocation = FilterWriter.class.getResource(FILTER_XSD_RESOURCE_PATH);
       SchemaFactory schemaFactory = SchemaFactory.newInstance(SCHEMA_LANGUAGE);
+      schemaFactory.setResourceResolver(
+          (type, namespaceURI, publicId, systemId, baseURI) -> {
+            if (!SCHEMA_LANGUAGE.equals(type)) {
+              return null;
+            }
+            URI remoteSchema = URI.create(systemId);
+            String protocol = remoteSchema.getScheme();
+            if (!HTTP.equals(protocol)) {
+              return null;
+            }
+            String path = remoteSchema.getPath();
+            String filename = path.substring(path.lastIndexOf('/') + 1);
+            URL remoteSchemaLocal = FilterWriter.class.getResource("/schemas/" + filename);
```

Comment:
❓ Should we memoize this to reduce I/O overhead? Or is the schema factory smart enough to handle that itself?

Reply:
Looks like the schema factory is compiling schema files in preparation for validation, so this loading should only occur once, at bundle start. 
https://docs.oracle.com/javase/8/docs/api/javax/xml/validation/SchemaFactory.html

Reply:
❓ No need to worry if get resource returns null? This will always for sure be a valid filename?

Reply:
Correct. Short of modifying the jar and removing the resources, you shouldn't be able to blow this up. The input to the function is constant.